co_sim_io/impl/data_container.hpp

```diff
@@ -110,27 +114,30 @@ class DataContainerRawMemory : public DataContainer<TDataType>
 {
 public:
     explicit DataContainerRawMemory(TDataType** ppData, const std::size_t Size)
-        : mppData(ppData), mSize(Size) {}
+        : mppData(ppData), mSize(Size), mCapacity(Size) {}
 
     std::size_t size() const override {return mSize;};
     void resize(const std::size_t NewSize) override
     {
-        if (NewSize > mSize) { // only increase the capacity if too small => same behavior as std::vector
-            if(mSize != 0) // Maybe is not null neigther allocated: double *data. Pooyan.
-                free(*mppData); // this is ok according to the standard, no matter if it is null or allocated //also check if using "std::"
-
-            *mppData = (TDataType *)malloc((NewSize)*sizeof(TDataType)); // TODO maybe use realloc? //also check if using "std::"
+        if (NewSize > mCapacity) { // only increase the capacity if too small => same behavior as std::vector
+            if (mCapacity == 0) {
+                *mppData = nullptr;
+            }
+            *mppData = (TDataType *)realloc(*mppData, (NewSize)*sizeof(TDataType));
             CO_SIM_IO_ERROR_IF_NOT(*mppData) << "Memory reallocation failed";
+
+            mCapacity = NewSize;
```

Comment:
@pooyan-dadvand now the behavior is exactly the same as for std::vector, no matter how it is resized

Reply:
If I recall correctly, in many implementations of `std::vector`, if there is a need to increase the capacity, they increase it to the twice of current size for efficiency reasons.

Reply:
To be more precise: it will resize to the `max(mCapacity*2, NewSize)`

Reply:
good point!
I will check if it is possible also for us and implement it :)

Reply:
I did it a bit simpler, but can easily be changed in the future

Reply:
edit I now do what you suggested, seems to work better